fix!: use platform name in snap filename (#5588)
snapcore/snapcraft: docs/release-notes/snapcraft-8-11.rst

.. _release-8.11:

Snapcraft 8.11 release notes
============================

> 4 August 2024

Reviewer: ```suggestion
4 August 2025
```

Learn about the new features, changes, and fixes introduced in Snapcraft 8.11.

Requirements and compatibility
------------------------------
See :ref:`reference-system-requirements` for information on the minimum hardware and
installed software.

Backwards-incompatible changes
------------------------------

The following changes are incompatible with previous versions of Snapcraft.

Snap file name changes
~~~~~~~~~~~~~~~~~~~~~~

> The file name for core24 snaps now use the platform name instead of
> the build-for architecture.

Reviewer 2: ```suggestion
The file names for core24 snaps are now derived from the name of the entry in the 
``platforms`` key instead of the value of its ``build-for`` key.
```

For example, consider the following snippet:

.. code-block:: yaml
  :caption: snapcraft.yaml

  name: hello-world
  version: 1.0
  platforms:
    rpi:
      build-on: [arm64]
      build-for: [arm64]
    rpi-debug:
      build-on: [arm64]
      build-for: [arm64]

Previously, Snapcraft would build both snaps and use the same name,
``hello-world_1.0_arm64.snap``, for both snaps. This would result in the first snap
being overwritten by the second snap.

> Now, Snapcraft builds 2 separate snaps, ``hello-world_1.0_rpi.snap`` and

Reviewer 2: ```suggestion
Now, Snapcraft builds two separate snaps, ``hello-world_1.0_rpi.snap`` and
```

https://docs.ubuntu.com/styleguide/en/#numbers

Author: https://github.com/canonical/snapcraft/pull/5588/commits/96aeed826009dba1fcf33ef667f3ff89301f5c8b
